modules/aws/src/smithy4s/aws/AwsClient.scala

```diff
@@ -21,30 +21,34 @@ import cats.effect.Resource
 import cats.syntax.all._
 
 import internals.AwsJsonRPCInterpreter
+import smithy4s.ShapeTag
+import smithy4s.ShapeId
 
 object AwsClient {
 
   def apply[Alg[_[_, _, _, _, _]], F[_]: MonadThrow](
       service: smithy4s.Service[Alg],
       awsEnv: AwsEnvironment[F]
   ): Resource[F, AwsClient[Alg, F]] =
-    prepare(service).map(_.build(awsEnv)).liftTo[Resource[F, *]]
+    prepare(service)
+      .leftWiden[Throwable]
+      .map(_.build(awsEnv))
+      .liftTo[Resource[F, *]]
 
   def prepare[Alg[_[_, _, _, _, _]]](
       service: smithy4s.Service[Alg]
-  ): Either[Throwable, AWSInterpreterBuilder[Alg]] =
+  ): Either[InitialisationError, AWSInterpreterBuilder[Alg]] =
```

Comment:
This whole change to typed errors may be overkill, let me know what you think. I'd say we should expose the public list of protocols anyway, but having them in the error type gives callers the ability to show better errors on their side as well.

Reply:
I'd rather not, on the account that it makes it harder to retain bincompat

Reply:
In dynamic land it's not completely certain that it happens at bootstrapping 😄

Reply:
sure, but at the same time in dynamic land you're likely to lift the `Either` into an `F` and let it propagate, aren't you ?

Reply:
not really, I have preparation as a separate step and I might want to show a different message depending on whether:

- the service is not an AWS service at all
- the service is AWS but we don't support the protocol yet

Reply:
@Baccata just reminding about this one - it's breaking territory so we should figure it out before 0.17

Reply:
Okay. The point that convinces me is that the SimpleRestBuilder does have a "UnsupportedProtocolError" typed error on the left. 

Rename the ADT `AwsClientInitialisationError` and move it to its own file and we'll merge